plasmapy/formulary/collisions.py

```diff
@@ -2104,6 +2104,56 @@ def coupling_parameter(
     return coulomb_energy / kinetic_energy
 
 
+def _get_arguments_not_provided(self, expected_arguments):
+    """
+    Collect parameters that weren't provided during instantiation needed
+    to access a specific attribute.
+    """
+
+    arguments_not_provided = []
+
+    for required_argument in expected_arguments:
+        if isinstance(required_argument, str):
+            if getattr(self, required_argument) is None:
+                arguments_not_provided.append(required_argument)
+        elif isinstance(required_argument, tuple):
+            number_of_both_or_either_provided = 0
+
+            for mutually_exclusive_argument in required_argument:
+                if getattr(self, mutually_exclusive_argument) is not None:
+                    number_of_both_or_either_provided += 1
+
+            if number_of_both_or_either_provided == 0:
+                arguments_not_provided.append("/".join(required_argument))
+
+    return arguments_not_provided
+
+
+def _validate_arguments(expected_arguments):
```

Comment:
Should I move this to somewhere in the utilities modules?

Reply:
Definitely more general than this file on collisions - is there nothing similar in the decorators file already? If not, my vote is to add them there. Not sure if they need separate tests there?

Reply:
The current validation decorators only work for functions, so I figured it was probably more simple to write a new decorator for attributes provided during class instantiation as opposed to trying to expand the functionality of the existing ones

Reply:
If it does something similar to an existing one for functions, maybe this one should have the same name but with '_class' at the end?